### PR TITLE
Notify CheckBox Foreground update after Color changed (not before)

### LINKED ISSUE #9404: CheckBox control Color property not rendering correctly
### Description

I have noticed an issue with the CheckBox control. When setting the Color property, the actual property value is changed, but the color change does not render onscreen. When the color is changed again, the previous color is used. And so on.

This is the current behavior on iOS/MacCatalyst and Android. On Windows, the color of the CheckBox doesn't change at all.

### Steps to Reproduce

Create a new page in a .NET MAUI app. Add the following to the XAML file (substituting your page class name):

    <HorizontalStackLayout VerticalOptions="Start">
        <Button x:Name="Button" Clicked="Button_Clicked"/>
        <CheckBox x:Name="Check" IsChecked="True"/>
    </HorizontalStackLayout>

Adjust the code behind file as follows:

    public partial class [class name] : ContentPage
    {
        bool isGreen;

        public [class name]()
        {
            InitializeComponent();
            SetControls();
        }

        void Button_Clicked(object sender, EventArgs e)
        {
            isGreen = !isGreen;
            SetControls();
        }

        void SetControls()
        {
            Check.Color = isGreen ? Colors.Green : Colors.Red;
            Button.TextColor = Check.Color;
            Button.Text = $"Is green? {isGreen}";
        }
    }

Run. Note that the CheckBox glyph is initially red. Press the button. As expected, the color and text of the button changes to green, however the color of the glyph remains red. Now press the button again. The button changes back to red, and this time the color of the CheckBox glyph turns green (when it should be red).

Also note that the property is being changed correctly: in SetControls(), the CheckBox color is set first, and the Button text color is set from CheckBox.Color. The issue is that the control color is not being updated.

### Version with bug

6.0.400

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, Windows, macOS

### Affected platform versions

iOS 15.5, Mac Catalyst 15.?, Android API 29 Win10-19041

### Did you find any workaround?

No

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Notify Foreground update after Color changed. We was invalidating the UpdateForeground method before update the Foreground property value.

fix-9404

### Issues Fixed

Fixes #9404

### CODE REVIEW

File: src/Controls/src/Core/HandlerImpl/CheckBox/CheckBox.Impl.cs [hunk 1]
```diff
@@ -12,6 +12,8 @@ public partial class CheckBox : ICheckBox, IMapColorPropertyToPaint
 		void IMapColorPropertyToPaint.MapColorPropertyToPaint(Color color)
 		{
 			Foreground = color?.AsPaint();
```
Comment (mattleibow) [inline src/Controls/src/Core/HandlerImpl/CheckBox/CheckBox.Impl.cs:14]: Doesn't this automatically trigger the Foreground property changed?

Comment (jsuarezruiz) [reply src/Controls/src/Core/HandlerImpl/CheckBox/CheckBox.Impl.cs:14]: Yes, but in the wrong order. This is thrown first, and then the property is updated.

Comment (mattleibow) [reply src/Controls/src/Core/HandlerImpl/CheckBox/CheckBox.Impl.cs:14]: Oh that interface looks special... I never saw it before. And if going the simpler and used in other places way works then maybe that is the best way for now?

Comment (jsuarezruiz) [reply src/Controls/src/Core/HandlerImpl/CheckBox/CheckBox.Impl.cs:14]: Just sent an alternative removing `IMapColorPropertyToPaint` here https://github.com/dotnet/maui/pull/12102

**Comment by jsuarezruiz** [comment]
Close this one in favor of https://github.com/dotnet/maui/pull/12102

**Pull request closed without merging**